olx_parser.py

```diff
@@ -1,124 +1,186 @@
-                soup_product_number_id = soup_product.find('span', class_='css-12hdxwj er34gjf0').text                 
-                soup_product_name_salesman = soup_product.find('h4', class_='css-1lcz6o7 er34gjf0').text
-
-                soup_product_date_registered = soup_product.find('div', class_='css-16h6te1 er34gjf0').text.split(' ')
-                date_registered = format_date(soup_product_date_registered)
-                
-
-                soup_product_date_of_last_visit = soup_product.find('span', class_='css-1t0qnkx').text.split(' ')
-                date_of_last_visit = format_date(soup_product_date_of_last_visit)
-                
-                soup_product_date_posted = soup_product.find('span', class_='css-19yf5ek').text.split(' ')
-                date_posted = format_date(soup_product_date_posted)
-                
-                soup_product_jpg = soup_product.find('img').get('src')                  
-                soup_product_title = soup_product.find('h4', class_='css-1juynto').text
-                soup_product_text = soup_product.find('div', class_='css-1t507yq er34gjf0').text
-        
-        except(requests.RequestException, ValueError, AttributeError):
-            print('error')
+            card_url = div.find('a').get('href')
+            if card_url:
+                if 'https://' in card_url:
+                    adt_urls.append(card_url)
+                else:
+                    adt_urls.append(f'https://www.olx.kz{card_url}')
+        except Exception:
+            print('error extract_adt_urls')
             continue
+    print(adt_urls)
+    return (adt_urls)
 
 
+def parser_adt(adv_html: str) -> Adv | None:
+    adv_html = BeautifulSoup(adv_html, 'html')
+    try:
+        soup_product_number_id = adv_html.find('span', class_='css-12hdxwj er34gjf0').text # ID объявления
+        soup_product_name_salesman = adv_html.find('h4', class_='css-1lcz6o7 er34gjf0').text # Ник продавца
+        soup_product_date_registered = adv_html.find('div', class_='css-16h6te1 er34gjf0').text 
+        date_registered = parse_register_date(soup_product_date_registered) # Дата регистрации аккаунта
+        soup_product_date_of_last_visit = adv_html.find('span', class_='css-1t0qnkx').text
+        date_of_last_visit = parse_last_online_date(soup_product_date_of_last_visit) # Время последнего посещения сайта
+        soup_product_date_posted = adv_html.find('span', class_='css-19yf5ek').text
+        date_posted = parse_published_date(soup_product_date_posted) # Дата размещения объявления
+        soup_product_jpg = adv_html.find('img').get('src') # Ссылка на фотографию объявления
+        soup_product_title = adv_html.find('h4', class_='css-1juynto').text # Название объявления
+        soup_product_text = adv_html.find('div', class_='css-1t507yq er34gjf0').text # Описание объявления   
+    except(requests.RequestException, ValueError, AttributeError):
+        print(f'{"_"*51}error adt{"_"*51}')
```

Comment:
Тут неплохо бы логировать информацию об ошибках, в идеале с трейсбэком, то есть используя logging. Да и в других местах тоже.
```suggestion
    except(requests.RequestException, ValueError, AttributeError) as err:
        logger.error(f'{"_"*51}error adt{"_"*51}',  exc_info=err)
```